### DISCUSSION TITLE
Support for native types passed as parameters

### DISCUSSION THREAD

**Discussion by skyhook88** [pacman82/arrow-odbc-py#46] [Ideas]
Noticed that the parameters are always handed and passed to database as strings.  Any plans to extend this?   So far it all seems to work with casting to string and I'm guessing this approach allows the solution to be more generic however it will pose an issue if needing to work with binary blobs etc.

**Comment by pacman82** [comment]
Casting everything to string allowed me to unlock a lot, but not all use cases. Passing typed parameters is of course possible, however so far the need has not been popped up. For inserts the payload of the arrays is already transported in typed buffers (kinda the point of this artefact). For the query part the most important reason to enable passing the arguments separately is to avoid SQL injection which is done reasonably well with string arguments.

However if a true use case pops up requiring binary blobs, or other type parameters for readers I do not see anything fundamentally wrong translating it to other types.

Specification would have to be done very careful though. Mapping from Python to Relational types, and deciding on their representation in C.
